sdk/filter/filter.go

```diff
@@ -0,0 +1,10 @@
+package filter
```

Comment:
Current types semantics seem ambiguous to me, does `block` mean `true` or `block` means `false`, could we use imperatives?

Reply:
I was chatting with @davexroth and there was a brief mention of wanting to keep the filter generic as in the future we will want to use it for more than just blocking.
Ah, the span should be there as argument, let me add that.

Reply:
These types return true or false so I don't think we can do more than block/don't block. If we want to add other kind of functionality we better add other types or interfaces. Imagine you want to cover other use cases than blocking (would be nice to know some), how would you differentiate those from the blocking to not block?

I truly think we should be explicit rather than cryptic as this is going to impact user workflow.

cc @davexroth

Reply:
Agreed normally we should have better return types, but in this case I'm ok with just returning true/false, as this is what's being implemented in the other environments and how our blocking implementation has been across the languages.

As Ryan mentioned, I do want to keep this interface consistent across languages - as the dream is to be able to have this one common interface across agents, and then we'd have a single webasam filter which could be loaded in any of the runtimes.  When the interface deviates between implementations, it makes that more difficult.